clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp

```diff
@@ -1024,10 +1029,27 @@ inline sycl::event matmul(descriptor_ptr handle, matmul_desc_ptr compute_desc,
     matmul_args.insert(
         {DNNL_ARG_ATTR_SCALES | DNNL_ARG_WEIGHTS, *scales_alpha});
   }
-
+  sycl::queue &queue = ::dpct::cs::get_default_queue();
   if (compute_desc->_epilogue != epilogue_t::nop) {
     ::dnnl::post_ops matmul_ops;
-    matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_relu, 0.f, 0.f);
+     if (compute_desc->_epilogue == epilogue_t::relu) {
+      matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_relu, 0.f, 0.f);
+    }  else if (compute_desc->_epilogue == epilogue_t::gelu) {
+      matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_gelu_erf, 0.f, 0.f);
+    }  else if (compute_desc->_epilogue == epilogue_t::gelu_aux) {
+      matmul_ops.append_eltwise(::dnnl::algorithm::eltwise_gelu_erf, 0.f, 0.f);
+      dpct::blas::matrix_mem_copy(compute_desc->_epilogue_aux_pointer, new_c,
```

Comment:
I forget one issue. This `dpct::blas::matrix_mem_copy` does not return sycl::event. So we cannot control the dependency chain when using non-USM.

Reply:
Yes correct, for the second point, I see it is raised in JIRA. Currently I could not find a way to extract immediate result of A*B+C for the aux memcpy. 

For the first case, do I have to force use USM always then ? What do you suggest ?

Reply:
I think we can do something like:
```
    sycl::event copy_e;
    size_t size_of_element =
        dpct::detail::library_data_size[static_cast<unsigned int>(
            d_desc->_type)] /
        8;
    if (compute_desc->_epilogue_aux_ld == d_desc->_ld) {
      size_t copy_size =
          size_of_element * ((n - 1) * (size_t)(d_desc->_ld) + m);
      copy_e = ::dpct::cs::memcpy(
          *q_ptr, (void *)(compute_desc->_epilogue_aux_pointer), (void *)new_d,
          copy_size, ::dpct::cs::memcpy_direction::automatic, {matmul_prim_event});
    } else {
      copy_e = ::dpct::cs::merge_events(
          *q_ptr,
          ::dpct::cs::memcpy(*q_ptr, compute_desc->_epilogue_aux_pointer, new_d,
                             size_of_element * compute_desc->_epilogue_aux_ld,
                             size_of_element * d_desc->_ld, size_of_element * m,
                             n, ::dpct::cs::memcpy_direction::automatic, {matmul_prim_event}));
    }
```
While currently the `::dpct::cs::memcpy` cannot accept dependency events and there is no `::dpct::cs::merge_events`.
We need also adjust/add those 2 APIs.

Reply:
Yes adjusted changes after discussion, please help to review. Thanks